Wolverine.Weather.API/Controllers/WeatherForecastController.cs

```diff
@@ -22,33 +22,34 @@ IMapper mapper
             _mapper = mapper;
         }
 
-        [HttpGet] //The get attribute tag does not need a name Ex. [HttpGet(Name = GetWeatherForecast)]. C# will name it after the controller by truncating off the "*Controller" part
-        public async Task<ActionResult<IEnumerable<WeatherForecast>>> GetAll(CancellationToken cancellationToken) //HOMEWORK 4.2.2024: Do this the correct way 
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<WeatherForecast>>> GetAll(CancellationToken cancellationToken)
         {
             var result = await _weatherForecastService.GetWeatherForecasts(cancellationToken);
             return Ok(result);
         }
 
         [HttpGet]
-        [Route("{id}")]
-        public async Task<ActionResult<WeatherForecast>> Get(int id, CancellationToken cancellationToken)
+        [Route("{ExternalId}")]
+        public async Task<ActionResult<WeatherForecast>> Get(Guid ExternalId, CancellationToken cancellationToken)
         {
-            if(id <= 0)
-            {
-                return BadRequest($"Invalid Id:{id}");
-            }
-            var result = await _weatherForecastService.GetWeatherForecast(id, cancellationToken);
+            //if(ExternalId <= 0)
```

Comment:
Is it possible to call this endpoint but not pass in a Guid, or is it possible to pass in something like '0', we might still need to do some validation to stop those cases

Reply:
No extra validation needed